repository/postgresql/user_test.go

```diff
@@ -0,0 +1,260 @@
+package repository_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/serhiihuberniuk/blog-api/models"
+	repository "github.com/serhiihuberniuk/blog-api/repository/postgresql"
+	"github.com/stretchr/testify/assert"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func createPostgresTestContainer(ctx context.Context) (testcontainers.Container, *pgxpool.Pool, error) {
+	dbName := "test_postgres_db"
+	port := "5432/tcp"
+	env := map[string]string{
+		"POSTGRES_PASSWORD": "password",
+		"POSTGRES_USER":     "postgres",
+		"POSTGRES_DB":       dbName,
+	}
+
+	req := testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:        "postgres",
+			Env:          env,
+			ExposedPorts: []string{port},
+			WaitingFor:   wait.ForListeningPort(nat.Port(port)),
+		},
+		Started: true,
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, req)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error occurred while creating container: %w", err)
+	}
+
+	mappedPort, err := container.MappedPort(ctx, nat.Port(port))
+	if err != nil {
+		return container, nil, fmt.Errorf("error occurred while getting port: %w", err)
+	}
+
+	dbUrl := fmt.Sprintf("postgres://postgres:password@localhost:%s/%s?sslmode=disable", mappedPort.Port(), dbName)
+
+	pool, err := repository.NewPostgresDb(ctx, dbUrl, "../../init.sql")
+	if err != nil {
+		return container, nil, fmt.Errorf("cannot create conn pool: %w", err)
+	}
+
+	return container, pool, nil
+}
+
+func cleanUserTable(t *testing.T, pool *pgxpool.Pool, ctx context.Context) {
+	t.Helper()
+
+	_, err := pool.Exec(ctx, "delete from users")
+	if err != nil {
+		t.Log(fmt.Errorf("cannot clean users table: %w", err))
+	}
+}
+
+func TestRepository_User(t *testing.T) {
+	ctx := context.Background()
+
+	container, pool, err := createPostgresTestContainer(ctx)
```

Comment:
please refactor function `createPostgresTestContainer` to return Repo, CloseFunc, error